apps/mobile/src/hooks/useNotificationPayload.ts

```diff
@@ -20,11 +21,18 @@ export function useNotificationPayload(appSigners: Record<string, AddressInfo>)
   const fcmToken = useAppSelector(selectFCMToken)
 
   const getNotificationRegisterPayload = useCallback(
-    async (nonce: string | undefined) => {
+    async ({
+      nonce,
+      ramdomPK
```

Comment:
I don't like that we pass the random PK here. Can't we pass a signer instead and move the calculation of the signer to the useRegisterForNotificaiton hook?

Having to find the owner of the safe and then going for a random PK if none is found is a weird thing for the getNotificationRegisterPayload function. 

this fucntion also returns accountType, but it doesn't seem that the account type is actually needed for the SiWE message, rather it is needed later when proposing to the backend. Same goes for the fcmToken and signer?

If you need those things up, you will have a pretty simple funciton signature
nonce, signer. and the function would return just the payload (which is currently implied by the function name)